src/Fantomas/Context.fs

```diff
@@ -1448,6 +1448,15 @@ let internal genTriviaBeforeClausePipe (rangeOfClause: range) ctx =
         | None -> id
     <| ctx
 
+let internal genTriviaMainNodesBeforeClausePipe (rangeOfClause: range) ctx =
+    (Map.tryFindOrEmptyList SynMatchClause_Clause ctx.TriviaMainNodes)
+    |> List.tryFind (fun t -> RangeHelpers.rangeEq t.Range rangeOfClause)
+    |> fun trivia ->
+        match trivia with
+        | Some trivia -> indent +> (printContentBefore trivia)
```

Comment:
`indent` without matching `unindent` is a very dangerous thing.
You need to add another test to verify that the indentation is correct again.
Something like:

```fsharp
[<Test>]
let ``foo `` () =
    formatSourceString false """
let outer =
    try
        //humberto
        TrySomething(someParam)
    with
        //comentario
        ex -> MakeSureToCleanup(someParam)

    let someBinding v = v + 2
"""  config
    |> prepend newline
    |> should equal """
let outer =
    try
        //humberto
        TrySomething(someParam)
    with
        //comentario
        ex -> MakeSureToCleanup(someParam)

    let someBinding v = v + 2
"""
```

But as mention above, I'd prefer to not have this helper function.

Reply:
@nojaf Thanks so much for your comments. I will work in that direction!